### PR TITLE
Fix clang-wrapper.sh so it handles arguments with spaces.

### PR DESCRIPTION
**Pull request by vargaz**
Fixes https://github.com/dotnet/runtime/issues/92335.

### CODE REVIEW

File: eng/clang-wrapper.sh [hunk 1]
```diff
@@ -22,4 +22,4 @@ case $CLANG_NAME in
         ;;
 esac
 
-$CLANG_CC $EXTRA_ARGS $@
+$CLANG_CC $EXTRA_ARGS "$@"
```
Comment (akoeplinger) [inline eng/clang-wrapper.sh:25]: should we wrap `CLANG_CC` as well, in case it has spaces?

```suggestion
"$CLANG_CC" $EXTRA_ARGS "$@"
```

**Review by akoeplinger — APPROVED** [review]

**Comment by akoeplinger** [comment]
I assume we need to backport this?

**Comment by vargaz** [comment]
The bug is visible with 8.0.

**Pull request merged**